core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -53,11 +55,12 @@
 
 import io.netty.channel.ChannelFuture;
 import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.EventLoop;
 import io.netty.handler.codec.http2.Http2Error;
 import io.netty.util.AsciiString;
 import io.netty.util.ReferenceCountUtil;
 
-final class HttpResponseSubscriber implements Subscriber<HttpObject>, RequestTimeoutChangeListener {
+final class HttpResponseSubscriber implements Subscriber<HttpObject>, TimeoutController {
```

Comment:
Do you think it's a good idea to make this class extend `DefaultTimeoutController`? Would it still hold when we add support for other timeouts, like intra-stream idle timeout?

Reply:
> Do you think it's a good idea to make this class extend DefaultTimeoutController? 

Initially, I made `HttpResponseSubscriber` extend `DefaultTimeoutController`.
While implementing this PR, I felt a little weird about directly extending `DefaultTimeoutController`. Because I thought `TimeoutController` is a kind of an auxiliary function.
On the other hand, this is a package-private class so simple implementation is good for maintaining.
Let me try to extend `DefaultTimeoutController`.

> Would it still hold when we add support for other timeouts, like intra-stream idle timeout?

That is a good point. I didn't think about yet... I think if we share a `TimeoutController` instance to HTTP/2 session-level, we can reset timeout `onNext()` event?